dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/WorkflowDAO.java

```diff
@@ -134,7 +137,24 @@ public Workflow findByPath(String path, boolean findPublished) {
             query.setParameter("workflowname", workflowname);
         }
 
-        return uniqueResult(query);
+        return list(query);
+    }
+
+    /**
+     * Finds a BioWorkflow or Service matching the given path.
+     *
+     * Initial implementation currently calls findByPath and filters the result; would ideally do it as a query with
+     * no filtering instead.
+     *
+     * @param path
+     * @param findPublished
+     * @param clazz
+     * @param <T>
+     * @return
+     */
+    public <T extends Workflow> Optional<T> findByPath(String path, boolean findPublished, Class<T> clazz) {
+        final List<Workflow> workflows = findByPath(path, findPublished);
+        return workflows.stream().filter(w -> w.getClass().equals(clazz)).map(clazz::cast).findFirst();
```

Comment:
Maybe this should throw an exception if the stream has more than one after the filter()